### DISCUSSION TITLE
Waveform selection and other CC parameters

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_basic_synth#114] [Q&A] [answered]
Hi marcel license,
How can I select different waveforms in this ESP32 basic Synth, is it using buttons? Or knobs? I made an Arduino based midi HAT for this ESP32 basic Synth, which sits on top of the ESP32 Audio Kit. I have looked your midi messages on z config file and tried to send the exact same midi massages but it seems like only 3-4 knobs while rotated seems to have some effect on the audio output others not, all the knobs sends messages successfully I have checked it with midi monitor.
The thing I want to know is how to change and select waveforms, what are the options I have for creating interesting sounds? Is buttons are used to send one time CC messages or toggle CC messages?

**Comment by marcel-licence** [comment]
same as https://github.com/marcel-licence/esp32_basic_synth/discussions/113

Hi,
to select different waveforms #USE_UNISON must not be active.
When looking at https://github.com/marcel-licence/esp32_basic_synth/blob/main/z_config.ino
you will find to linked parameters:
```
    { 0x0, 0x10, "R1", NULL, Synth_SetParam, SYNTH_PARAM_WAVEFORM_1},
    { 0x1, 0x10, "R2", NULL, Synth_SetParam, SYNTH_PARAM_WAVEFORM_2},
```
You can also make a direct call:
```
Synth_SetParam(SYNTH_PARAM_WAVEFORM_1, value_1);
Synth_SetParam(SYNTH_PARAM_WAVEFORM_2, value_2);
```
to select a waveform from another function for example when using external knobs or pots.

In case you are using ADC_TO_MIDI_ENABLED you can put in the parameters to link to the waveform selection:
```
    {0, 0x10},
    {1, 0x10},
```

Hope this helps.

**Reply by souravlal** [reply]
Hi marcel,
I have not defined #USE_UNISON, but still no changes while turning the pots. 
Aren't R1, R2 normal potentiometers? Or is it rotary encoders? I am using normal pots.
And also no changes while using the buttons. Are the buttons send CC messages or Toggle CC messages?

**Reply by souravlal** [reply, marked as answer]
Hi marcel,
Finally I figured it out I was sending cc messages over the wrong channel. Now everything is working correctly. Thanks.

**Reply by marcel-licence** [reply]
Good to hear! Thank you very much for your feedback
